src/include/daos_obj.h

```diff
@@ -841,6 +841,18 @@ daos_obj_query_key(daos_handle_t oh, daos_handle_t th, uint64_t flags,
 int
 daos_obj_verify(daos_handle_t coh, daos_obj_id_t oid, daos_epoch_t epoch);
 
+/**
+ * Set an anchor for enumeration for one shard only.
+ *
+ * \param[in]	oh	Object open handle.
+ * \param[in/out]
+ *		anchor	Anchor modified for 1 shard only
+ *
+ * \return		0		Success and consistent
+ */
+int
+daos_obj_anchor_split(uint32_t shard, daos_anchor_t *anchor);
```

Comment:
What I had in mind was more something like this:
```
/**
  * Generate an array of anchors that split the dkey address space for parallel enumeration.
  *
  * \param[in]	oh	Object open handle.
  * \param[in/out]
  *                    nr	Size of anchors array. If 0 is passed, return the optimal size back.
  *                    	 Number of split anchors returned. 
  * \param[in/out]
  *		 anchors	Array of split anchors
  *
  * \return ...
  */
 int
 daos_obj_anchor_split(daos_handle_t oh, uint32_t *nr, daos_anchor_t *anchors);
```

Reply:
hmm, but how would you distribute the array of anchors to everyone?
The way it works today is that one guy get the optimal number of anchors, and inserts works into the queue for each shard. otherwise I will probably need to modify this patch more to allow an anchor to span a range but not all shards, which isn't supported today.